pkg/operation/hybridbotanist/cloud_config.go

```diff
@@ -55,12 +56,29 @@ func (b *HybridBotanist) generateCloudConfigChart() (*chartrenderer.RenderedChar
 		cloudProvider["config"] = cloudProviderConfig
 	}
 
+	machineTypes := b.Shoot.GetMachineTypesFromCloudProfile()
+	memoryThreshold, _ := resource.ParseQuantity("8Gi")
+
 	workers := []map[string]interface{}{}
-	for _, workerName := range userDataConfig.WorkerNames {
-		workers = append(workers, map[string]interface{}{
-			"name":       workerName,
-			"secretName": b.Shoot.ComputeCloudConfigSecretName(workerName),
-		})
+
+	for _, worker := range b.Shoot.GetWorkers() {
+		newWorker := map[string]interface{}{
+			"name":       worker.Name,
+			"secretName": b.Shoot.ComputeCloudConfigSecretName(worker.Name),
+		}
+		for _, machtype := range machineTypes {
+			if machtype.Name == worker.MachineType {
+				if machtype.Memory.Cmp(memoryThreshold) > 0 {
+					newWorker["evictionHard"] = "1Gi"
```

Comment:
we should put that under the `memory.available` key somehow, right? it's not clear for what this value is used.

Reply:
yes, sounds good